File: Plus One/PlusOne.cpp
#include<iostream>
#include<vector>
using namespace std;

vector <int> plusOne(vector<int> &digits){
    for (int i = digits.size() - 1; i >= 0; i--){
        if (digits[i] == 9)    digits[i] = 0;
        else{
            digits[i]++;
            return digits;
        }
    }

    digits[0] = 1;
    digits.push_back(0);
    return digits;
}

int main(){
    vector <int> digits = {8, 4, 5, 9};
    vector <int> result = plusOne(digits);
    for (int i = 0; i < result.size(); i++){
        cout << result[i] << " ";
    }

    return 0;
}

File: Plus One/PlusOne.py
def plusOne(digits: list[int]) -> list[int]:
    for i in range(len(digits)-1, -1, -1):
        if digits[i] == 9:
            digits[i] = 0
        else:
            digits[i] = digits[i] + 1
            return digits
    return [1] + digits
    
   
if __name__ == "__main__": 
    a = [1,2,9]
    b = plusOne(a)
    print(b)